WaterwitchGUI/src/main.cpp

```diff
@@ -0,0 +1,256 @@
+                        }
+                        std::ofstream ofs;
+                        ofs.open("configs/" + string(config.name) + ".json", std::ofstream::out | std::ofstream::trunc);
+                        ofs << configJson.dump();
+                        ofs.close();
+                    }
+                    ImGui::EndTabItem();
+                }
+                ImGui::EndTabBar();
+            }
+            ImGui::End();
+        }
+
+        //camera window
+        if (showCameraWindow) {
+            ImGui::SetNextWindowSize(ImVec2(io.DisplaySize.x, (io.DisplaySize.y - ImGui::GetFrameHeight())));
+            ImGui::SetNextWindowPos(ImVec2{0, ImGui::GetFrameHeight()});
+            ImGui::Begin("Camera Window", &showCameraWindow, ImGuiWindowFlags_NoResize | ImGuiWindowFlags_NoScrollbar | ImGuiWindowFlags_NoScrollWithMouse);
+
+            ImVec2 windowPos = ImGui::GetWindowPos();
+            ImVec2 availPos = ImGui::GetContentRegionAvail();
+
+            ImGui::SetCursorPos(ImVec2(windowPos.x+8, windowPos.y+8));
+            ImGui::Button("Camera 1", ImVec2((availPos.x-24)/2, (availPos.y-24)/2));
+            ImGui::SetCursorPos(ImVec2((availPos.x-24)/2+16, windowPos.y+8));
+            ImGui::Button("Camera 2", ImVec2((availPos.x-24)/2, (availPos.y-24)/2));
+            ImGui::SetCursorPos(ImVec2(windowPos.x+8, (availPos.y-24)/2+35));
+            ImGui::Button("Camera 3", ImVec2((availPos.x-24)/2, (availPos.y-24)/2));
+
+            ImGui::End();
+        }
+
+        //co-pilot controls window
+        if (showPilotWindow) {
+            ImGui::Begin("Co-Pilot Window", &showPilotWindow);
+
+            ImGui::SliderInt("Surge", &power.surge, 0, 100);
+            ImGui::SliderInt("Sway", &power.sway, 0, 100);
+            ImGui::SliderInt("Turn", &power.turn, 0, 100);
+            ImGui::SliderInt("Heave", &power.heave, 0, 100);
+
+            ImGui::SeparatorText("Keybinds");
+            ImGui::Text("1 - Set all to 0%");
+            ImGui::Text("2 - Set all to 50%");
+            ImGui::Text("3 - Set all to 0%, set Heave to 100%");
+
+            if (ImGui::IsKeyPressed(ImGuiKey_1)) {
+                power.surge = 0;
+                power.sway = 0;
+                power.turn = 0;
+                power.heave = 0;
+            }
+            if (ImGui::IsKeyPressed(ImGuiKey_2)) {
+                power.surge = 50;
+                power.sway = 50;
+                power.turn = 50;
+                power.heave = 50;
+            }
+            if (ImGui::IsKeyPressed(ImGuiKey_3)) {
+                power.surge = 0;
+                power.sway = 0;
+                power.turn = 0;
+                power.heave = 100;
+            }
+
+            ImGui::End();
+        }
+
+        glClear(GL_COLOR_BUFFER_BIT);
+
+        ImGui::Render();
+        ImGui_ImplOpenGL3_RenderDrawData(ImGui::GetDrawData());
+
+        glfwSwapBuffers(window);
+    }
```

Comment:
I really like how these windows look and how everything is done. It seems it will be pretty easy to port this to `gui.cpp` in `ROS2/ROS_workspace/src/cli_app/src`.